### PR TITLE
tmux-sessionizer 0.4.2 (new formula)

### PR DESCRIPTION
**Pull request by junglerobba**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by carlocab — COMMENTED** [review]
Thanks for the PR. A few suggestions/comments.

File: Formula/t/tmux-sessionizer.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  on_linux do
+    depends_on "zlib"
+  end
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_match "tmux-sessionizer #{version}", shell_output("#{bin}/tms --version")
```
[the remark this answers is not in the document]
Comment (junglerobba) [reply Formula/t/tmux-sessionizer.rb]: First of all, thanks for the review!
It's a mostly interactive tool, so I tried to find a slightly better test case.

**Review by carlocab — APPROVED** [review]
Thanks!

File: Formula/t/tmux-sessionizer.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
+class TmuxSessionizer < Formula
+  desc "Tool for opening git repositories as tmux sessions"
+  homepage "https://github.com/jrmoulton/tmux-sessionizer/"
+  url "https://github.com/jrmoulton/tmux-sessionizer/archive/refs/tags/v0.4.2.tar.gz"
+  sha256 "0f9369a045ebe181202fcf5f292bbdd836f25b47ca9da1702351a725693631f5"
+  license "MIT"
+
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  depends_on "libgit2@1.7"
```
Comment (chenrui333) [inline Formula/t/tmux-sessionizer.rb:10]: I saw this pr https://github.com/jrmoulton/tmux-sessionizer/pull/104 has updated to use libgit2 1.8, any timeline to release that change?

**Review by chenrui333 — APPROVED** [review]

File: Formula/t/tmux-sessionizer.rb (hunk 2 above)
Comment (junglerobba) [reply Formula/t/tmux-sessionizer.rb:10]: I asked for a new release a while back here: https://github.com/jrmoulton/tmux-sessionizer/issues/123
The maintainer doesn't seem too active lately, aside from merging one PR recently

File: Formula/t/tmux-sessionizer.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    binary.dynamically_linked_libraries.any? do |dll|
+      next false unless dll.start_with?(HOMEBREW_PREFIX.to_s)
+
+      File.realpath(dll) == File.realpath(library)
+    end
+  end
+
+  test do
+    assert_match "Configuration has been stored", shell_output("#{bin}/tms config -p /dev/null")
+    assert_match version.to_s, shell_output("#{bin}/tms --version")
```
Comment (junglerobba) [inline Formula/t/tmux-sessionizer.rb:36]: output of the version command looks like this
```suggestion
    assert_match "tmux-sessionizer ${version}", shell_output("#{bin}/tms --version")
```

**Review by carlocab — APPROVED** [review]

**Pull request merged**